modules/pam_userdb/pam_userdb.c

```diff
@@ -209,36 +209,48 @@ user_lookup (pam_handle_t *pamh, const char *database, const char *cryptmode,
 	  char *cryptpw = NULL;
 
 	  if (data.dsize < 13) {
-	    compare = -2;
+	    /* hash from db too short */
+	    pam_syslog(pamh, LOG_INFO, "password hash in database too short");
 	  } else if (ctrl & PAM_ICASE_ARG) {
-	    compare = -2;
+	    pam_syslog(pamh, LOG_INFO,
+	       "case-insensitive comparison only works with plaintext passwords");
 	  } else {
+	    /* libdb is not guaranteed to produce null terminated strings */
+	    char *pwhash = strndup(data.dptr, data.dsize);
+
+	    if (pwhash == NULL) {
+	      pam_syslog(pamh, LOG_INFO, "strndup failed: data.dptr");
+        } else {
 #ifdef HAVE_CRYPT_R
-	    struct crypt_data *cdata = NULL;
-	    cdata = malloc(sizeof(*cdata));
-	    if (cdata != NULL) {
-		cdata->initialized = 0;
-		cryptpw = crypt_r(pass, data.dptr, cdata);
-	    }
+	      struct crypt_data *cdata = NULL;
+	      cdata = malloc(sizeof(*cdata));
+	      if (cdata == NULL) {
+	        pam_syslog(pamh, LOG_INFO, "malloc failed: struct crypt_data");
+          } else {
```

Comment:
Wrong indentation on this line

Reply:
Thanks, I found one more wrongly indented one and fixed. Unrelated question - would you accept a PR fixing the indentation of the whole file? If so, what are the guidelines?

Reply:
That's the problem. There are no guidelines for code format in Linux-PAM. So I'd recommend not doing that PR. At least not until we establish some official code format guideline for Linux-PAM.

Reply:
Up until now it was something like - follow the existing code format in the file you edit.  Of course in case some file is just a mess of different code formats we would probably even without a guideline accept a PR that would unify the format within the file to the code format the majority of the file is already in.

Reply:
Thanks for explaining. I won't hujer into things.